### PR TITLE
Fix Unit Test General Flakiness

### PR DESCRIPTION
**Pull request by cicoyle**
Working to fix the flakiness in unit tests in CI. They pass locally, so need to test a few things in CI since the behavior is different.

### CODE REVIEW

**Comment by cicoyle** [comment]
Looks like the last unit test failure is due to the membership test, [which is fixed here](https://github.com/dapr/dapr/pull/7817)

File: pkg/operator/api/informer/informer_test.go [hunk 1]
```diff
@@ -75,6 +75,11 @@ func Test_WatchUpdates(t *testing.T) {
 			},
 			operator.ResourceEventType_UPDATED,
 		)
+
+		require.Eventually(t, func() bool {
+			return i.batchID == 1
+		}, 5*time.Second, 100*time.Millisecond)
```
Comment (JoshVanL) [inline pkg/operator/api/informer/informer_test.go:81]: We want to print out the values if it does fail. This also has a data race so we need to change `i.batchID` to be a `atomic.Uint32.

```suggestion
		require.EventuallyWithT(t, func(c *assert.CollectT) {
			assert.Equal(t, 1, i.batchID)
		}, 5*time.Second, 100*time.Millisecond)
```

File: pkg/operator/api/informer/informer_test.go [hunk 2]
```diff
@@ -86,6 +91,11 @@ func Test_WatchUpdates(t *testing.T) {
 			},
 			operator.ResourceEventType_UPDATED,
 		)
+
+		require.Eventually(t, func() bool {
+			return i.batchID == 2
```
Comment (JoshVanL) [inline pkg/operator/api/informer/informer_test.go:96]: Same here.

File: pkg/operator/api/informer/informer_test.go [hunk 3]
```diff
@@ -94,6 +104,10 @@ func Test_WatchUpdates(t *testing.T) {
 			operator.ResourceEventType_CREATED,
 		)
 
+		require.Eventually(t, func() bool {
+			return i.batchID == 3
```
Comment (JoshVanL) [inline pkg/operator/api/informer/informer_test.go:108]: Same here.

**Review by JoshVanL — CHANGES_REQUESTED** [review]

**Comment by cicoyle** [comment]
it looks like the memory utilization actually decreased, but idk why the go routines upped by 1 for this diff:
`75th percentile for number of go routines (new): 37.00`

File: pkg/diagnostics/resiliency_monitoring_test.go [hunk 4]
```diff
@@ -407,7 +407,7 @@ func TestResiliencyActivationsCountMonitoring(t *testing.T) {
 					})
 				}
 				// let the circuit breaker to go to half open state (5x cb timeout) and then return success to close it
-				time.Sleep(500 * time.Millisecond)
+				time.Sleep(1000 * time.Millisecond)
```
Comment (cicoyle) [inline pkg/diagnostics/resiliency_monitoring_test.go:410]: I think its just a timing issue for this: https://github.com/dapr/dapr/actions/runs/9570009389/job/26384075690?pr=7818

**Review by JoshVanL — APPROVED** [review]

**Pull request merged**